src/index.ts

```diff
@@ -323,7 +317,52 @@ class Logify<ErrorTypes = ErrorParamType> {
       console.log(...logMessage)
     }
 
-    this._sendLogToGrafana(this._constructBackendLogEvent('fatal', message, params).jsonLog)
+    this._sendLogToGrafana(this._constructBackendLogEvent('fatal', message, params))
+  }
+
+  /**
+   * Stores an event (due to auth token missing or no network connection) to be sent to Grafana later, using "sendStoredLogs" function.
+   * @param log `string` - JSON.stringify-ed log object
+   * @private
+   */
+  private _store = (log: string) => {
+    if (!this.props.storage) {
+      return
+    }
+
+    const { storageKey, storage } = this.props
+
+    const current = JSON.parse(storage.getItem(storageKey)) || []
+    storage.setItem(storageKey, JSON.stringify([...current, log]))
```

Comment:
since both `getItem()` and `setItem()` can be either sync or async, we should use a specific api in this file that allows us to invoke the methods right away if they are sync or await them if they are async.

eg:

```ts
// works regardless of getItem() being sync or async
const data = await Promise.resolve(storage.getItem(key))
```

```ts
// works regardless of setItem() being sync or async
await Promise.resolve(storage.setItem(key, JSON.stringify([...current, log])))
```

Reply:
Ok, gotcha. Except there's no need to wrap it in a Promise.resolve(), I think.
It should work as long as it's await'ed.
AFAIK, if it's sync, it will resolve immediately, otherwise it will await the result of (set/get)Item and assign it to `data`.

Reply:
that was mainly to please typescript as awaiting a non-promise should return some angry warnings/errors 😅

Reply:
Nope, it was actually the other way around. The warnings were thrown once wrapped in a redundant Promise.resolve().